core/auth/fake/identifier.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	fakeConfig struct {
+		Broker           string                `json:"broker"`
+		LoginTemplate    string                `json:"loginTemplate"`
+		ValidatePassword bool                  `json:"validatePassword"`
+		UsernameFieldID  string                `json:"usernameFieldId"`
+		PasswordFieldID  string                `json:"passwordFieldId"`
+		UserConfig       map[string]userConfig `json:"userConfig"`
+	}
+
+	userConfig struct {
+		Password string `json:"password"`
+	}
+
+	viewData struct {
+		FormURL    string
+		Message    string
+		UsernameID string
+		PasswordID string
+	}
+)
+
+const (
+	defaultLoginTemplate = `
+<body>
+  <h1>Login!</h1>
+  <form name="fake-login-form" action="{{.FormURL}}" method="post">
+	<div>{{.Message}}</div>
+	<label for="{{.UsernameID}}">Username</label>
+	<input type="text" name="{{.UsernameID}}" id="{{.UsernameID}}">
+	<label for="{{.PasswordID}}">Password</label>
+    <input type="password" name="{{.PasswordID}}" id="{{.PasswordID}}">
+	<button type="submit" id="submit">Fake Login</button>
+  </form>
+</body>
+`
+
+	defaultUserNameFieldID = "username"
+	defaultPasswordFieldID = "password"
+
+	userDataSessionKey = "core.auth.fake.%s.data"
+)
+
+var (
+	_ auth.RequestIdentifier = (*identifier)(nil)
+	_ auth.WebCallbacker     = (*identifier)(nil)
+	_ auth.WebLogouter       = (*identifier)(nil)
+
+	errMissingUsername  = errors.New("missing username")
+	errInvalidUser      = errors.New("invalid user")
+	errMissingPassword  = errors.New("missing password")
+	errPasswordMismatch = errors.New("password mismatch")
+)
+
+// IdentityProviderFactory -
+func IdentityProviderFactory(cfg config.Map) (auth.RequestIdentifier, error) {
+	var fakeConfig fakeConfig
+	if err := cfg.MapInto(&fakeConfig); err != nil {
+		return nil, err
+	}
+
+	return &identifier{broker: fakeConfig.Broker, config: fakeConfig}, nil
+}
+
+// Inject injects module dependencies
+func (i *identifier) Inject(
+	reverseRouter web.ReverseRouter,
+	responder *web.Responder,
+	eventRouter flamingo.EventRouter,
+) *identifier {
+	i.reverseRouter = reverseRouter
+	i.responder = responder
+	i.eventRouter = eventRouter
+
+	return i
+}
+
+// Broker returns the broker id from the config
+func (i *identifier) Broker() string {
+	return i.broker
+}
+
+// Authenticate action, fake
+func (i *identifier) Authenticate(_ context.Context, r *web.Request) web.Result {
+	var formError error
+
+	//r.Request().G
+	if r.Request().Method == http.MethodPost {
+		postValues, err := r.FormAll()
+		if err == nil {
+			delete(postValues, "broker")
+			if len(postValues) > 0 {
+				formError = i.handlePostValues(r, postValues, i.broker)
+
+				if formError == nil {
+					return i.responder.RouteRedirect("core.auth.callback", map[string]string{"broker": i.broker})
+				}
+			}
+		}
+	}
+
+	return i.prepareFormResponse(formError, r)
+}
+
+func (i *identifier) prepareFormResponse(formError error, r *web.Request) web.Result {
+	// get formURL to callback with broker filled in
```

Comment:
The url points to core.auth.login, not core.auth.callback

Reply:
thx - fixed